test/cases/NoUselessIfElse.webidl

```diff
@@ -0,0 +1,46 @@
+[Exposed=Window]
+interface NoUselessIfElse {
+  // # Overloads:
+  undefined overloadsObjectOrBoolean(object arg1);
+  undefined overloadsObjectOrBoolean(boolean arg1);
+
+  undefined overloadsObjectOrBooleanOrNumeric(object arg1);
+  undefined overloadsObjectOrBooleanOrNumeric(boolean arg1);
+  undefined overloadsObjectOrBooleanOrNumeric(long arg1);
+
+  undefined overloadsObjectOrNumeric(object arg1);
+  undefined overloadsObjectOrNumeric(long arg1);
+
+  undefined overloadsBooleanOrNumeric(boolean arg1);
+  undefined overloadsBooleanOrNumeric(long arg1);
+
+  // ## Avoid regressions:
```

Comment:
I thought the present bug has to do with union types? Is overloading affected by the bug as well?

I'd also prefer a more surgical test suite. The snapshot generated from this file is really huge and not reviewable.

Reply:
> Is overloading affected by the bug as well?

The current version isn’t, because overloads had the bug fixed at some point.